spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/MappingMongoConverter.java

```diff
@@ -2006,12 +2006,16 @@ public boolean isAssignableFrom(org.springframework.data.util.TypeInformation ta
 		/**
 		 * Converts a source object into {@link TypeInformation target}.
 		 *
-		 * @param source must not be {@literal null}.
+		 * @param source can be {@literal null}.
 		 * @param typeHint must not be {@literal null}.
 		 * @return the converted object.
 		 */
 		@SuppressWarnings("unchecked")
-		public <S extends Object> S convert(Object source, TypeInformation<? extends S> typeHint) {
+		public <S extends Object> S convert(@Nullable Object source, TypeInformation<? extends S> typeHint) {
+
+			if(source == null) {
```

Comment:
Shouldn't that be rather caught in the map and collection iterations as we basically make `convert` nullable?

Reply:
The `MappingMongoConverter` does not follow the `o.s.c.c.converter.Converter` contract and accepted `null` in previous versions, though the documentation did state it differently. Since it's a public API, I'd rather restore the original behaviour and alter the documentation, than spread `null` checks in various places.